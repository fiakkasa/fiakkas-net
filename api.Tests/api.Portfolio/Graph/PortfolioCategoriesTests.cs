using HotChocolate.Execution;

namespace api.Portfolio.Tests;

public class PortfolioCategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task PortfolioCategories_Should_Return_Results()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  portfolioCategories(where: { version: { eq: 1 } }) {
    items {
      createdAt
      href
      id
      title
      updatedAt
      version
      portfolioItems {
        pageInfo {
          hasNextPage
          hasPreviousPage
        }
        items {
          createdAt
          href
          id
          ordinal
          title
          updatedAt
          version
          year
        }
      }
    }
    pageInfo {
      hasNextPage
      hasPreviousPage
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
