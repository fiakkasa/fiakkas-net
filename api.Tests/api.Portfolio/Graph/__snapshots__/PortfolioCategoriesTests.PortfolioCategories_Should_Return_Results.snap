{
  "data": {
    "portfolioCategories": {
      "items": [
        {
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "00000000-0000-0000-0000-000000000000",
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "portfolioItems": {
            "pageInfo": {
              "hasNextPage": false,
              "hasPreviousPage": false
            },
            "items": [
              {
                "createdAt": "2024-01-01T00:00:00.000Z",
                "href": "/test",
                "id": "00000000-0000-0000-0000-000000000000",
                "ordinal": 0,
                "title": "Title",
                "updatedAt": null,
                "version": 1,
                "year": 0
              }
            ]
          }
        }
      ],
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      }
    }
  }
}
